client/lib/utils/coerce-values.js

```diff
@@ -0,0 +1,85 @@
+import coerce from 'type-coerce';
+
+/**
+ * Retrieve a field's schema, handling referenced schema definitions if need be.
+ *
+ * @param {Object} fieldSchema - Schema object for field.
+ * @param {Object} definitions - Common definitions.
+ * @returns {Object} - Schema object for field, potentially pulled from $ref's definition.
+ */
+const getFieldSchema = ( fieldSchema, definitions ) => {
+	if ( fieldSchema.hasOwnProperty( '$ref' ) ) {
```

Comment:
This `hasOwnProperty` thing always seemed overly-verbose to me. Wouldn't it be easier to just check if `fieldSchema.$ref` is `truthy`? Same thing applies to your other uses of `hasOwnProperty`.